src/zope/i18nmessageid/tests.py

```diff
@@ -15,146 +15,202 @@
+        self.assertEqual(message.domain, 'domain')
+        self.assertEqual(message.default, 'default')
+        self.assertEqual(message.mapping, mapping)
+        self.assertEqual(message.msgid_plural, 'testings')
+        self.assertEqual(message.default_plural, 'defaults')
+        self.assertEqual(message.number, 2)
+        if self._TEST_READONLY:
             self.assertTrue(message._readonly)
 
-    def test_ctor_explicit(self):
+    def test_values_with_float_for_number(self):
         mapping = {'key': 'value'}
-        message = self._makeOne('testing', 'domain', 'default', mapping)
+        message = self._makeOne(
+            'testing', 'domain', 'default', mapping,
+            msgid_plural='testings', default_plural="defaults", number=2.2)
         self.assertEqual(message, 'testing')
         self.assertEqual(message.domain, 'domain')
         self.assertEqual(message.default, 'default')
         self.assertEqual(message.mapping, mapping)
-        if self._TEST_REAOONLY:
+        self.assertEqual(message.msgid_plural, 'testings')
+        self.assertEqual(message.default_plural, 'defaults')
+        self.assertEqual(message.number, 2.2)
+        if self._TEST_READONLY:
             self.assertTrue(message._readonly)
 
-    def test_ctor_copy(self):
+    def test_copy(self):
         mapping = {'key': 'value'}
-        source = self._makeOne('testing', 'domain', 'default', mapping)
+        source = self._makeOne(
+            'testing', 'domain', 'default', mapping,
+            msgid_plural='testings', default_plural="defaults", number=2)
         message = self._makeOne(source)
         self.assertEqual(message, 'testing')
         self.assertEqual(message.domain, 'domain')
         self.assertEqual(message.default, 'default')
         self.assertEqual(message.mapping, mapping)
-        if self._TEST_REAOONLY:
+        self.assertEqual(message.msgid_plural, 'testings')
+        self.assertEqual(message.default_plural, 'defaults')
+        self.assertEqual(message.number, 2)
+        if self._TEST_READONLY:
             self.assertTrue(message._readonly)
 
-    def test_ctor_copy_w_overrides(self):
+    def test_copy_with_overrides(self):
         mapping = {'key': 'value'}
         source = self._makeOne('testing')
-        message = self._makeOne(source, 'domain', 'default', mapping)
+        message = self._makeOne(
+            source, 'domain', 'default', mapping,
+            msgid_plural='testings', default_plural="defaults", number=2)
         self.assertEqual(message, 'testing')
         self.assertEqual(message.domain, 'domain')
         self.assertEqual(message.default, 'default')
         self.assertEqual(message.mapping, mapping)
-        if self._TEST_REAOONLY:
+        self.assertEqual(message.msgid_plural, 'testings')
+        self.assertEqual(message.default_plural, 'defaults')
+        self.assertEqual(message.number, 2)
+        if self._TEST_READONLY:
             self.assertTrue(message._readonly)
```

Comment:
There should be one more test, where both source and message's `__init__`  get all the attributes, and we get to see which ones get used in the end.

We'll then see if I'm right that C and Python choose different priorities.